lib/reek/configuration/configuration_file_finder.rb

```diff
@@ -31,7 +31,7 @@ def find_by_cli(options)
 
       def find_by_dir(start)
         start.ascend do |dir|
-          files = dir.children.select(&:file?)
+          files = dir.children.select(&:file?).sort
```

Comment:
I think `Pathname#children` returns the files in filesystem order, which is probably inode-based, so removing `exceptions.reek` and adding it back pushed it to the end of the returned children.